polylang/polylang
# Remove DOMNodeInserted depreciation

Fixes https://github.com/polylang/polylang-pro/issues/741

The `DOMNodeInerted` mutation event is deprecated and according to MDN it should be replaced by a `MutationObserver` (see: https://developer.mozilla.org/en-US/docs/Web/API/MutationEvent).

Also see https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe for the observer configuration.

⚠️ It appears that the language dropdown is available for default category. It seems that the `default_cat_{$term_id}` identifiers is not printed on the server side. Will be fixed in another PR.

## Changes:
- Introduce a mutation observer in both `post.js` an `term.js`. The only places where `DOMNodeInserted` was used.
- Check for `childList` mutation on the post list and term list tables.
- When a new node is inserted, assert it is the quick edit form before performing any treatment.

## Test instructions:
- Visit the quick edit in posts and terms.
- All should operates as before. Meaning that categories, tags and parent pages selectors should be filtered by language for posts and pages.
- Language dropdown should display the corresponding language of the currently edited post or term.
## To go further:
It would be interesting to remove the dependency to jQuery here and make a full use of ESNext syntax.
